SIMElasticity.h

```diff
@@ -390,6 +392,27 @@ template<class Dim> class SIMElasticity : public Dim
         continue;
       else if (!strcasecmp(child->Value(),"texturematerial"))
       {
+        if (Dim::dimension != 2)
+        {
+          std::cerr << "Texture material not supported for trivariate models" << std::endl;
+          return false;
+        }
+        for (size_t i=1; i<=this->getNoPatches(); ++i)
+        {
+          ASMs2D *patch = dynamic_cast<ASMs2D*>(this->getPatch(i));
+          if (!patch)
+          {
+            std::cerr << "Only works for ASMs2D..." << std::endl;
+            return false;
+          }
+          Go::SplineSurface *surf = patch->getSurface();
+          if (surf->startparam_u() != 0 || surf->endparam_u() != 1 ||
+              surf->startparam_v() != 0 || surf->endparam_v() != 1)
+          {
+            std::cerr << "Texture material requires unit parametric domain" << std::endl;
+            return false;
+          }
+        }
```

Comment:
Don't think you should add these limitations here. Essentially, you only want to make sure the parameters (u,v) always are in the domain [0,1]. Then it is better to implement this check inside the `IsotropicTextureMat::findMaterial` method instead, which is where the restriction actually is located.

Reply:
I can't put it inside the `IsotropicTextureMat` because it doesn't have access to the parametric range umin/umax. I can't just tag it as an error outside [0,1] because I also want spline patches with `umin=0.5`, `umax=0.7` to fail. This is the only place I found that I could put the test.

That said, I hope that this should be a temporary thing. As I said in the PR text I don't really want these restrictions here at all, I just couldn't engineer my way around them. Having explicit verbose error messages was the best I could do for iteration 1.